packages/neovim/src/utils/findNvim.ts

```diff
@@ -170,14 +189,22 @@ function getPlatformPaths() {
  * @param opt.minVersion See {@link FindNvimOptions.minVersion}
  * @param opt.orderBy See {@link FindNvimOptions.orderBy}
  * @param opt.firstMatch See {@link FindNvimOptions.firstMatch}
+ * @param opt.paths See {@link FindNvimOptions.paths}
+ * @param opt.dirs See {@link FindNvimOptions.dirs}
  */
 export function findNvim(opt: FindNvimOptions = {}): Readonly<FindNvimResult> {
-  const paths = getPlatformPaths();
+  const platformDirs = getPlatformSearchDirs();
+  const nvimExecutable = windows ? 'nvim.exe' : 'nvim';
+
+  const allPaths = new Set<string>([
+    ...(opt.paths ?? []).map(normalizePath),
+    ...(opt.dirs ?? []).map(dir => normalizePath(join(dir, nvimExecutable))),
+    ...[...platformDirs].map(dir => join(dir, nvimExecutable)),
+  ]);
 
   const matches = new Array<NvimVersion>();
   const invalid = new Array<NvimVersion>();
-  for (const path of paths) {
-    const nvimPath = join(path, windows ? 'nvim.exe' : 'nvim');
+  for (const nvimPath of allPaths) {
     if (existsSync(nvimPath)) {
```

Comment:
oh, this was for the case where we were searching directories. in that case, we still want to ignore "not found" because we are manually building the path by appending `nvim`.

but for the case where `paths` is explicitly passed, I think it makes sense to add any/all of those paths to `invalid` if they are not found.

Reply:
For "not found", what error should we put in error field in invalid?